static/src/javascripts/projects/common/modules/article/spacefinder.js

```diff
@@ -147,12 +155,14 @@ define([
         if (rules.selectors) {
             forOwn(rules.selectors, function (params, selector) {
                 filtered = filtered.then(function (slots) {
+                    return [slots, qwery(rules.bodySelector + selector)];
+                }).then(function (args) {
```

Comment:
I don't know how good you already understand the code but I feel like this file should probably be the most well commented/documented piece of functionality. I can imagine that the next person in 6 months looking at this will have hard times again to understand what and especially why exactly this piece of code is doing. It might not be the task for you for this PR but we should finally find time to document it properly. Unless you feel it is documented well enough already.